numpy/core/numeric.py

```diff
@@ -401,11 +402,17 @@ def asarray(a, dtype=None, order=None):
         Input data, in any form that can be converted to an array.  This
         includes lists, lists of tuples, tuples, tuples of tuples, tuples
         of lists and ndarrays.
-    dtype : data-type, optional
+    dtype : {None, data-type}, optional
         By default, the data-type is inferred from the input data.
-    order : {'C', 'F'}, optional
-        Whether to use row-major ('C') or column-major ('F' for FORTRAN)
-        memory representation.  Defaults to 'C'.
+    order : {'C', 'F', 'A'}, optional
+        Whether to use row-major ('C'), column-major ('F'), or any ('A')
+        memory representation.  Defaults to 'A'.
+    precision: {None, dtype, ndarray, scalar}, optional
```

Comment:
Yeah, let's not overload one kwarg with multiple semantics like that...

On Thu, Mar 5, 2015 at 1:19 PM, Charles Harris <notifications@github.com>
wrote:

> In numpy/core/numeric.py
> <https://github.com/numpy/numpy/pull/5634#discussion_r25902442>:
>
> >          By default, the data-type is inferred from the input data.
> > -    order : {'C', 'F'}, optional
> > -        Whether to use row-major ('C') or column-major ('F' for FORTRAN)
> > -        memory representation.  Defaults to 'C'.
> > +    order : {'C', 'F', 'A'}, optional
> > +        Whether to use row-major ('C'), column-major ('F'), or any ('A')
> > +        memory representation.  Defaults to 'A'.
> > +    precision: {None, dtype, ndarray, scalar}, optional
>
> Hmm, that suggests another approach, which is to reuse the dtype keyword
> and add a flag for interpretation, something like min_dtype=True. I'm not
> a big fan of entangled keywords though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numpy/numpy/pull/5634/files#r25902442>.
>



-- 
Nathaniel J. Smith -- http://vorpus.org